### ISSUE TITLE
Why is the 'output. jpg' only a black image?

### ISSUE THREAD

**Issue by johoye123** [zcemycl/TF2DeepFloorplan#85]
Hi:

After downloading the code from the Git repository:

Using the command 
python3 -m dfp.train --batchsize=4 --lr=5e-4 --epochs=100 --logdir=log/store --modeldir=model/store and r3d.tfrecords datasets, successfully generate files such as log/store/G and model/store/savd_model.pb;

But using the command 
python3 -m src.dfp.deploy --image resources/123.jpg --weight src/log/store/G --postprocess --colorize --save output. jpg -- loadmethod log, the generated output. jpg is a completely black image?

Any help is appreciated.
Thanks in advance

**Comment by johoye123** [comment]
The following is the print information for the deploy command:

python3 -m src.dfp.deploy --image resources/123.jpg --weight src/log/store/G --postprocess --colorize --save output.jpg --loadmethod log
2024-08-07 20:03:20.574143: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-08-07 20:03:20.630728: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2024-08-07 20:03:20.631155: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-08-07 20:03:21.230581: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2024-08-07 20:03:23.066238: E tensorflow/compiler/xla/stream_executor/cuda/cuda_driver.cc:268] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
/opt/opencv/TF2DeepFloorPlan/TF2DeepFloorplan/src/dfp/utils/util.py:34: UserWarning: ignoring size because footprint is set
  ndimage.grey_erosion(output_array, size=(3, 3), footprint=el),
(1024, 1024, 3) (1024, 1024, 3)
(1024, 1024, 3)

**Comment by zcemycl** [comment]
The logs you printed out look fine to me. It only indicates you have no GPU or not installed GPU library cuda correctly, but it is fine it can use cpu for prediction. 

About why black image, this is a hobbyist repo, i don't guarantee perfect training and prediction on your image. And training a Deep Learning model does not mean you always getting the same or good result, thats why you need to monitor your loss graph by tensorboard.

**Comment by johoye123** [comment]
OK， Thank you very much for replying to me
